llama-index-core/llama_index/core/node_parser/relational/markdown_element.py

```diff
@@ -57,7 +57,7 @@ def get_nodes_from_node(self, node: TextNode) -> List[BaseNode]:
         self.extract_table_summaries(table_elements)
         # convert into nodes
         # will return a list of Nodes and Index Nodes
-        return self.get_nodes_from_elements(elements)
+        return self.get_nodes_from_elements(elements, node.id_)
```

Comment:
So its not really a filename, but assumes the node id is file name?

Should we just instead inherit all metadata?

Reply:
`node.id_` is a filename if node parser is getting documents from local files.

for ``` inherit all metadata```, do you mean we could inherit all the metadata from base nodes to the inherited nodes. but base nodes sometime has  just empty metadata and id_ is filename. so that is the only information useful for inherited nodes. 

sorry, maybe have a quick call to understand this better